src/full/Agda/TypeChecking/Forcing.hs

```diff
@@ -176,15 +176,21 @@ nextIsForced (f:fs) = (f, fs)
 forcingTranslation :: [NamedArg DeBruijnPattern] -> TCM [NamedArg DeBruijnPattern]
 forcingTranslation ps = do
   (qs, rebind) <- dotForcedPatterns ps
-  case rebind of
-    Nothing -> return ps
-    Just rebind -> do
+  -- If qs == ps, no patterns were dotted.  Otherwise, report result.
+  verboseS "tc.force" 50 $
+    unless (qs == ps) $
       reportSDoc "tc.force" 50 $ "forcingTranslation" <?> vcat
         [ "patterns:" <?> pretty ps
         , "dotted:  " <?> pretty qs
-        , "rebind:  " <?> pretty rebind ]
+        , "rebind:  " <?> pretty rebind
+        ]
+  -- If @rebind@ is empty, there is no further work to do.
+  -- Still, @qs@ might be different from @ps@:
+  -- Dots might have been moved, just not onto binders or strict matches.
```

Comment:
Motivating why it's ok to stop if `rebind` is empty:

`dotForcedPatterns` adds dot patterns and `rebindForcedPattern` turns to dot patterns into proper patterns. If `rebind` is empty we might have created some new dot patterns, but since we won't remove any dot patterns there is no need to have another go at `dotForcedPatterns`.

Reply:
I'll add your comment to the commit.